src/Symfony/Bundle/FrameworkBundle/Test/WebTestAssertionsTrait.php

```diff
@@ -183,21 +183,27 @@ public static function assertRouteSame($expectedRoute, array $parameters = [], s
             $constraint = LogicalAnd::fromConstraints($constraint, ...$constraints);
         }
 
-        self::assertThat(static::getRequest(), $constraint, $message);
+        self::assertThat(self::getRequest(), $constraint, $message);
     }
 
-    private static function getClient(): KernelBrowser
+    private static function getClient(KernelBrowser $newClient = null): ?KernelBrowser
```

Comment:
Why would it return `null`? If the client is null, it should alway throw an exception, shouldn't it? If it's null, all the rest of the trait's codebase would be forced to check for null pointer 😕

Reply:
related to my answer to your other comment: `getClient(null)` returns `null`